### PR TITLE
feat: add activity-log skill for end-of-day recap (#1198)

### LINKED ISSUE #1198: Add read-only activity-log skill so the end-of-day recap can surface autonomous actions
## Context

With auto-RSVP (and other autonomous medium-risk actions) now possible in v0.38, the **end-of-day "work Curia did" recap** should surface them (e.g. "RSVP'd accept to John Doe's invite, booked Tue"). The daily *digest* is the wrong home — it's forward-looking ("what the CEO has coming up"); the recap is the right home.

Every skill execution — including each `calendar-respond-to-invite` RSVP — is already persisted to the append-only `audit_log` table via a write-ahead `skill.result` bus event (`src/audit/logger.ts:99`). **But no skill exposes `audit_log` to agents.** Available action skills are `list-pending-actions` (pending only), `approve`/`deny`/`dismiss-action`, and `scheduler-report` (scheduled jobs, not in-turn skill executions). So the recap job currently has no read path to "what autonomous actions did I take."

## Proposed fix — read-only activity/action-history skill

Add a read-only skill (`action_risk: none`), e.g. `list-recent-actions` / `activity-log`, that queries `audit_log` for completed actions over a time window, filterable by action class / skill name / agent, returning a structured list (`timestamp`, `skill`, `target`, `outcome`) suitable for a recap. Then pin it to the recap-capable agent and point the existing recap job at it.

Design notes:
- Filter to consequential actions (e.g. `action_risk` ≥ low, or an allowlist of skills like `calendar-respond-to-invite`) so the recap isn't flooded with `none`-risk reads.
- Parameterized queries only; respect the append-only / read-only nature of `audit_log`.
- Consider distinguishing autonomous (bypassed/auto) vs approved actions if that data is in the payload.

## Acceptance criteria

- [ ] A read-only skill returns recent completed actions from `audit_log` for a given time window (e.g. last 24h), with skill name, target, outcome, and timestamp.
- [ ] The skill is `action_risk: none`, uses parameterized queries, and performs no writes.
- [ ] Auto-RSVPs (`calendar-respond-to-invite` executions) appear in the result and can be filtered for.
- [ ] The skill is pinned to the recap agent; the end-of-day recap job's prompt is updated to query and report autonomous actions.
- [ ] Unit test with seeded `audit_log` rows covering an RSVP action.

---
**Follow-up cluster:** v0.38 calendar scheduling-consult hardening (Cluster C — autonomous-action observability).

### PR DESCRIPTION
**Pull request by josephfung**
## Summary

Closes #1198

Adds a read-only `activity-log` skill so the end-of-day debrief can surface autonomous actions Curia took during the day (e.g. auto-RSVPs).

- **`AuditLogRepo`** — parameterized queries against `audit_log` for `skill.result` rows in a time window
- **`activity-log` skill** — `action_risk: none`, returns summarized actions (`timestamp`, `skill`, `target`, `outcome`, `detail`, `autonomy`); defaults to consequential skills (including `calendar-respond-to-invite`); optional `skill_name` filter
- **Coordinator** — pinned + debrief prompt instructs calling `activity-log` before composing the recap
- **Registry** — enabled by default on fresh installs

## Acceptance criteria

- [x] Read-only skill returns recent completed actions from `audit_log` for a time window
- [x] `action_risk: none`, parameterized queries, no writes
- [x] Auto-RSVPs appear and can be filtered via `skill_name`
- [x] Pinned to coordinator; debrief prompt updated
- [x] Unit test + integration test with seeded RSVP audit row

<div><a href="https://cursor.com/agents/bc-22b1619d-9aa9-43d5-8bdc-7ac38aa4e30d"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-web-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-web-light.png">Open in Web</picture></a>&nbsp;<a href="https://cursor.com/background-agent?bcId=bc-22b1619d-9aa9-43d5-8bdc-7ac38aa4e30d"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-cursor-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-cursor-light.png">Open in Cursor</picture></a>&nbsp;</div>

### CODE REVIEW

File: skills/activity-log/handler.ts [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+
+import type { SkillHandler, SkillContext, SkillResult } from '../../src/skills/types.js';
+import type { AuditLogRow } from '../../src/audit/audit-log-repo.js';
+import type { ActionLogRow } from '../../src/autonomy/action-log-types.js';
+import { toLocalIso, formatDisplayTimezone } from '../../src/time/timestamp.js';
+
+const ISO_DATETIME_RE = /^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}(?:\.\d{1,3})?(?:Z|[+-]\d{2}:\d{2})$/;
+
+/** Skills whose completed executions belong in a CEO activity recap by default. */
+const DEFAULT_RECAP_SKILLS = new Set([
```
Comment (josephfung) [inline skills/activity-log/handler.ts]: Silent-empty hazard. When `skill_name` is not supplied, results are filtered to this hardcoded set (applied at line 83). If a future consequential skill is added or renamed (a new `calendar-*` or outbound skill) and not added here, the recap silently omits it with zero signal, exactly the 'recap returns nothing when it should not' failure #1198 calls out. It also duplicates the `action_risk` classification already on each manifest. #1198 suggested filtering by `action_risk >= low` as the primary approach with the allowlist as fallback; recommend deriving the default filter from `action_risk` instead of a static name list, or at minimum add a comment plus a test asserting the set stays in sync.

File: skills/activity-log/handler.ts [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+    skill: skillName,
+    agent_id: row.sourceId,
+    target: extractTarget(skillName, result),
+    outcome: success ? 'completed' : 'failed',
+    detail: success ? extractSuccessDetail(skillName, result?.data) : (result?.error ?? null),
+    autonomy,
+  };
+}
+
+function matchAutonomyOutcome(
```
Comment (josephfung) [inline skills/activity-log/handler.ts:115]: This correlates an `audit_log` row to an `autonomy_action_log` row purely by `skillName` plus a +/- 5-min window, with no conversation or task key. Two same-skill actions within 5 minutes (e.g. two RSVPs) will both match the first row found, and an approved-then-re-executed action also lands as `outcome === 'success'`, so it gets labelled `autonomous` rather than `approved`, the exact distinction the coordinator prompt is told to surface. It falls back to `unknown` honestly when nothing matches, so this is a soft CEO-facing label rather than a data bug. Both tables carry `conversation_id`; joining on that would remove the ambiguity.

**Review by josephfung — COMMENTED** [review]
**Review summary, APPROVE WITH NITS** (posted as a comment; GitHub blocks approving your own PR.)

The cleanest of the batch structurally. The `activity-log` skill is read-only by construction (`action_risk: none`, no writes, routed through a sanctioned repo), the SQL is parameterized throughout (including the `= ANY($n::text[])` array filter and dynamically-indexed params), the manifest is loader-clean (`capabilities: [auditLogRepo, actionLogRepo]` are both gated services now in `VALID_CAPABILITIES`; no universal services declared; name matches directory), and registration is fully wired: added to `config/registry-defaults.yaml` and pinned in `coordinator.yaml`, with the repo plumbed end-to-end through `index.ts` to `ExecutionLayer` to `SkillContext`. Summarizing payloads instead of returning raw audit rows is a nice touch for avoiding payload leakage. No critical issues.

Two real, low-blast-radius quality items, both inline:
1. The default `DEFAULT_RECAP_SKILLS` allowlist silently drops anything not on it (`handler.ts:14`, applied at `:83`), the exact silent-empty failure #1198 warns about.
2. The autonomy-outcome enrichment can mis-attribute on a +/- 5-min, skill-name match with no conversation key (`handler.ts:131`).

Note for prod: per the registry-defaults mechanism, this auto-enrolls on fresh installs but will NOT retroactively enable on already-provisioned instances; it must be manually enabled in the prod `skill_registry`. Flagging the state, not a defect.

Test coverage is solid for the size (unit + integration). Minor gaps: no test for the `since`/`until` validation branches, the half-open `[since, until)` boundary, or the autonomy mis-match edge case.

File: skills/activity-log/handler.ts (hunk 1 above)
Comment (josephfung) [reply skills/activity-log/handler.ts]: Fixed in cffca92f. Replaced the hardcoded set with getRecapEligibleSkillNames() (src/skills/recap-skills.ts), which scans manifests at first use and includes any skill with action_risk >= low (min score 60). New consequential skills are picked up automatically; memory-query / activity-log (none) stay excluded. Unit test in recap-skills.test.ts.

File: skills/activity-log/handler.ts (hunk 2 above)
Comment (josephfung) [reply skills/activity-log/handler.ts:115]: Fixed in cffca92f. matchAutonomyOutcome now requires conversation_id to match when both the audit row and autonomy row carry one, prefers approved over success when multiple candidates exist in the window, and falls back to unknown on mismatch. Tests cover approved-by-conversation and mismatch → unknown.
